cmd/utils/encodedProgress.go

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the cpackget project. */
+
+package utils
+
+import (
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type EncodedProgress struct {
+	mu             sync.Mutex
+	total          int64
+	current        int
+	currentPercent int
+	instanceNo     int
+	name           string
+}
+
+func NewEncodedProgress(max int64, instNo int, filename string) *EncodedProgress {
+	return &EncodedProgress{
+		total:      max,
+		instanceNo: instNo,
+		name:       filename,
+	}
+}
+
+func (p *EncodedProgress) Add(count int) int {
+	p.mu.Lock()
+	newCount := count
+	p.current += newCount
+	p.Print()
+	p.mu.Unlock()
+	return newCount
+}
+
+func (p *EncodedProgress) Write(bs []byte) (int, error) {
+	p.mu.Lock()
+	newCount := len(bs)
+	p.current += newCount
+	p.Print()
+	p.mu.Unlock()
+	return newCount, nil
+}
```

Comment:
This is nearly identical code as above in `Add` function. Why?
Can this be written as:

```suggestion
func (p *EncodedProgress) Write(bs []byte) (int, error) {
	return p.Add(len(b)), nil
}
```